homeassistant/components/esphome/strings.json

```diff
@@ -130,6 +130,25 @@
     "service_calls_not_allowed": {
       "title": "{name} is not permitted to perform Home Assistant actions",
       "description": "The ESPHome device attempted to perform a Home Assistant action, but this functionality is not enabled.\n\nIf you trust this device and want to allow it to perform Home Assistant action, you can enable this functionality in the options flow."
+    },
+    "device_conflict": {
+      "title": "Device conflict for {name}",
+      "fix_flow": {
+        "step": {
+          "start": {
+            "title": "Device conflict for {name}",
+            "description": "The ESPHome device, `{name}` (`{model}`) at `{ip}` has changed from `{stored_mac}` to `{mac}`. If you have multiple devices with the same name, please rename the one with MAC address `{mac}` to a new name to avoid conflicts.\n\nIf you have replaced the device with new hardware, confirm that you want to migrate the configuration to the new device."
+          },
+          "confirm": {
+            "title": "Confirm device replacement for {name}",
+            "description": "Are you sure you want to migrate `{name}` (`{model}`) at `{ip}` from `{stored_mac}` to `{mac}`?"
```

Comment:
Probably should say Home Assistant to migrate its configuration so it's more clear that it's only in the context of HA